fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/physical/PruneShuffleColumnRule.java

```diff
@@ -112,6 +113,16 @@ private void prune(DistributionContext childContext) {
                 return;
             }
 
+            // check if any one of them are duplicated columns like [1,1] = [17,29]
```

Comment:
Like [a,b,b] = [c,d,e], I think we also could prune the columns

Reply:
in theory yes, but there is degraded case.